### PR TITLE
Renovation: refactor options cache in Widget

### PR DESCRIPTION
**Pull request by LazyLahtak**
Move logic of initializing, setting and getting options cache from `Widget` into `Options`. 
`Options` now have `cache(name, option)` method (works like methods `option(options, value)` and `silent(options, value)`)

### CODE REVIEW

**Comment by LazyLahtak** [comment]
Ready to review

**Review by churkin — DISMISSED** [review]

File: js/core/options/index.js [hunk 1]
```diff
@@ -197,4 +199,14 @@ export class Options {
     isDeprecated(name) {
         return Object.prototype.hasOwnProperty.call(this._deprecated, name);
     }
+
+    cache(name, options) {
```
Comment (IlyaKhD) [inline js/core/options/index.js:203]: Why not split into two simple methods? E.g. `addToCache` \ `readFromCache` or simpler `cahe` \ `fromCache`

Comment (IlyaKhD) [reply js/core/options/index.js:203]: No need for this `if(isGetter) {` logic, simpler to read - good points only 😉

Comment (LazyLahtak) [reply js/core/options/index.js:203]: We discussed this with @churkin and desided to make this method like `option` and `silent` methods.
Should we change them too?

Comment (IlyaKhD) [reply js/core/options/index.js:203]: No you shouldn't
I don't see how these methods are related to each other actually. `cache` is very simple and should not be over-weighted. `addToCache` and `getFromCache` is very clear, no one is going to be mixed up, I think.

Comment (LazyLahtak) [reply js/core/options/index.js:203]: Methods `option`, `silent` and `cache` work like getter and setter at same time. They also look very similar. I think we should leave them as they are now or refactor them all.

File: js/ui/widget/ui.widget.js [hunk 2]
```diff
@@ -179,30 +179,23 @@ const Widget = DOMComponentWithTemplate.inherit({
         this._initContentReadyAction();
     },
 
-    _clearInnerOptionCache(optionContainer) {
-        this[`${optionContainer}Cache`] = {};
-    },
+    _getOptionsFromContainer: function({ name, fullName, value }) {
+        var options = {};
```
Comment (DokaRus) [inline js/ui/widget/ui.widget.js]: Use `const` or `let` instead of `var`

Comment (LazyLahtak) [reply js/ui/widget/ui.widget.js]: Done in #10987

**Review by DokaRus — APPROVED** [review]

**Review by churkin — APPROVED** [review]

File: js/core/options/index.js (hunk 1 above)
Comment (churkin) [reply js/core/options/index.js:203]: You should not merge PR if it has not resolved comments.

Comment (LazyLahtak) [reply js/core/options/index.js:203]: Decided to leave them as they are now

**Pull request merged**